src/main/build.rs

```diff
@@ -88,11 +102,21 @@ fn run_cbindgen(build_common: &ShadowBuildCommon) {
     cbindgen::Builder::new()
         .with_crate(crate_dir)
         .with_config(cbindgen::Config {
+            // We want to avoid including any C headers from this crate here,
+            // which lets us avoid circular dependencies.
+            includes: vec![],
             include_guard: Some("main_opaque_bindings_h".into()),
-            no_includes: true,
             export: cbindgen::ExportConfig {
                 include: vec!["QDiscMode".into()],
-                item_types: vec![cbindgen::ItemType::OpaqueItems, cbindgen::ItemType::Enums],
+                // Export everything except function definitions, since those are already
+                // exported in the other header file, and need the C header files.
+                item_types: base_config
+                    .export
+                    .item_types
+                    .iter()
+                    .cloned()
+                    .filter(|t| *t != cbindgen::ItemType::Functions)
+                    .collect(),
                 ..base_config.export.clone()
             },
```

Comment:
No, we want the struct definitions for e.g. `SysCallReg` to be exported here. It's ok to depend on C headers and their struct definitions, just not shadow headers, since the latter leads to circular include dependencies.

Maybe it's confusing that I kept the `-opaque.h` suffix, but it wasn't immediately obvious what else to call these.